File: apps/notification/src/functions/NotificationAuthorizer.ts
'use strict'

import {Handler} from "aws-lambda";

const AbstractAuthorizer = require('../../../../shared/common/authorization/AbstractAuthorizer')
const {AuthPolicy, HttpVerb} = require('../../../../shared/common/authorization/AuthPolicy')

class NotificationAuthorizer extends AbstractAuthorizer {
    generatePermissions(authPolicy: typeof AuthPolicy, userId: string) {
        if (userId) {
            authPolicy.allowMethod(HttpVerb.POST, `/v1/users/${userId}/notifications/token`)
        } else {
            authPolicy.denyAllMethods()
        }
    }
}

const authorizer = new NotificationAuthorizer()

export const authorize: Handler = async (event) => {
    return await authorizer.authorize(event)
}

File: apps/notification/src/functions/NotificationAuthorizer.py
from src.functions.AbstractAuthorizer import AbstractAuthorizer
from src.functions.AuthPolicy import AuthPolicy, HttpVerb


class NotificationAuthorizer(AbstractAuthorizer):
    def __init__(self):
        AbstractAuthorizer.__init__(self)

    def generate_permissions(self, auth_policy: AuthPolicy, user_id: str):
        if user_id is not None:
            auth_policy.allow_method(HttpVerb.POST, f"/v1/users/{user_id}/notifications/token")
        else:
            auth_policy.deny_all_methods()


notification_authorizer = NotificationAuthorizer()


def authorize(event, context):
    return notification_authorizer.authorize(event)
